File: 0242-valid-anagram/solution.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if(s.size() != t.size()) return false;
        map<char, int> mpp;
        for(char ch : s){
            mpp[ch]++;
        }

        for(char ch: t){
            if(mpp[ch]) mpp[ch]--;
            else return false;
        }
        return true;

    }
};

File: 0242-valid-anagram/solution.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if(s.length() != t.length()) return false;
        int sMap[] = new int[26];

        for(int i=0; i<s.length(); i++){
            sMap[s.charAt(i) - 97]++;
        }

        for(int i=0; i<t.length(); i++){
            if(sMap[t.charAt(i) - 97] >= 1)
               sMap[t.charAt(i) - 97]--;
            else return false;
        }

        return true;
    }
}
